File: src/test/java/uk/gov/justice/hmpps/prison/service/TestClock.java
package uk.gov.justice.hmpps.prison.service;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;

import java.time.Clock;
import java.time.LocalDateTime;

@TestConfiguration
public class TestClock {
    @Bean
    public Clock clock() {
        return Clock.fixed(
            LocalDateTime.of(2020, 1, 2, 3, 4, 5).atZone(java.time.ZoneId.systemDefault()).toInstant(),
            java.time.ZoneId.systemDefault());
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/service/TestClock.kt
package uk.gov.justice.hmpps.prison.service

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import java.time.Clock
import java.time.LocalDateTime
import java.time.ZoneId

@TestConfiguration
class TestClock {
  @Bean
  fun clock(): Clock = Clock.fixed(
    LocalDateTime.of(2020, 1, 2, 3, 4, 5).atZone(ZoneId.systemDefault()).toInstant(),
    ZoneId.systemDefault(),
  )
}
